internal/transfer/transfer.go

```diff
@@ -99,13 +101,42 @@ func readBlockOpResponse(r io.Reader) (*hdfs.BlockOpResponseProto, error) {
 	return resp, err
 }
 
+const HOPSFS_CLOUD_DATANODE_HOSTNAME_OVERRIDE_ENV = "HOPSFS_CLOUD_DATANODE_HOSTNAME_OVERRIDE"
+const HOPSFS_CLOUD_DATANODE_PORT_OVERRIDE_ENV = "HOPSFS_CLOUD_DATANODE_PORT_OVERRIDE"
+
 func getDatanodeAddress(datanode *hdfs.DatanodeIDProto, useHostname bool) string {
 	var host string
 	if useHostname {
 		host = datanode.GetHostName()
 	} else {
 		host = datanode.GetIpAddr()
 	}
+	port := datanode.GetXferPort()
+
+	//Uncomment for debugging
```

Comment:
maybe add a debug flag instead and only run this if debug

Reply:
this lib does not have any mechanism for printing debug messages. I do not want to add a debug mechanism for this change.  I can add it later if needed.